fe/fe-core/src/main/java/com/starrocks/analysis/CreateRoutineLoadStmt.java

```diff
@@ -583,6 +643,135 @@ public static void analyzeCustomProperties(Map<String, String> dataSourcePropert
         }
     }
 
+    private void checkPulsarProperties() throws AnalysisException {
+        Optional<String> optional = dataSourceProperties.keySet().stream()
+                .filter(entity -> !PULSAR_PROPERTIES_SET.contains(entity))
+                .filter(entity -> !entity.startsWith("property.")).findFirst();
+        if (optional.isPresent()) {
+            throw new AnalysisException(optional.get() + " is invalid pulsar custom property");
+        }
+
+        // check service url
+        pulsarServiceUrl = Strings.nullToEmpty(dataSourceProperties.get(PULSAR_SERVICE_URL_PROPERTY)).replaceAll(" ", "");
+        if (Strings.isNullOrEmpty(pulsarServiceUrl)) {
+            throw new AnalysisException(PULSAR_SERVICE_URL_PROPERTY + " is a required property");
+        }
+
+        if (!Pattern.matches(ENDPOINT_REGEX, pulsarServiceUrl)) {
```

Comment:
The pulsar service url is in `pulsar://localhost:6550,localhost:6651,localhost:6652` style. This check seems useless.

Reply:
Right, I should check them one by one.